src/s11/main.c

```diff
@@ -200,6 +195,8 @@ s11_reader()
 
 		if(len > 0) {
 			MsgBuffer* tmp_buf_p = createMsgBuffer(len);
+			uint32_t ip = ntohl(g_s11_cp_addr.sin_addr.s_addr);
+			MsgBuffer_writeUint32(tmp_buf_p, ip, true);
```

Comment:
I am not sure what advantage we get by keeping it in the msgbuf ? If we keep it at message specific then code becomes intuitive and readable ? Your view ?

Reply:
Just added the source address of the message in case we needed it in future.

Reply:
This is the place where we have access to the source ip address of the message. g_s11_cp_addr is global, and after this the message is being put into a job queue. In case where multiple sgws are used, by the time previous message is handled, g_s11_cp_addr would have been overwritten by another ip. That is why the ip is written to the msg buffer at the time the message is received. 
@badhrinathpa can you clarify if my understanding of your previous comment is correct?